server/index.controller.js

```diff
@@ -15,4 +15,18 @@ function getEndpoints (req, res, next) {
   res.status(200).send({endpoints})
 }
 
-module.exports = {getTopics, getEndpoints}
\ No newline at end of file
+function getArticleById (req, res, next) {
+  const articleId = req.params.article_id
+  selectArticleById(articleId)
+  .then((article) => {
+    if (article.rows.length === 0) {
+      next({status: 404, msg: "not found"})
+    }
```

Comment:
This does work, but I would personally recommend having this type of error handling in the model file instead.